Modules/Microsoft365DSC/DSCResources/MSFT_AADUser/MSFT_AADUser.psm1

```diff
@@ -462,51 +462,59 @@ function Set-TargetResource
         }
         #endregion
 
-        if ($null -ne $Password)
+        if ($user.UserPrincipalName)
         {
-            $passwordValue = $Password.GetNetworkCredential().Password
+            Write-Verbose -Message "Updating Office 365 User $UserPrincipalName Information"
+
+            if ($null -ne $Password)
+            {
+                Write-Verbose -Message "PasswordProfile property will not be updated"
+            }
+
+            $CreationParams.Add('UserId', $UserPrincipalName)
+            Update-MgUser @CreationParams
         }
         else
         {
-            try
+
+            if ($null -ne $Password)
             {
-                # This only works in PowerShell 5.1
-                $passwordValue = [System.Web.Security.Membership]::GeneratePassword(30, 2)
+                $passwordValue = $Password.GetNetworkCredential().Password
             }
-            catch
+            else
             {
-                $TokenSet = @{
-                    U = [Char[]]'ABCDEFGHIJKLMNOPQRSTUVWXYZ'
-                    L = [Char[]]'abcdefghijklmnopqrstuvwxyz'
-                    N = [Char[]]'0123456789'
-                    S = [Char[]]'!"#$%&''()*+,-./:;<=>?@[\]^_`{|}~'
+                try
+                {
+                    # This only works in PowerShell 5.
+                    Add-Type -AssemblyName System.Web
+                    $passwordValue = [System.Web.Security.Membership]::GeneratePassword(30, 2)
                 }
+                catch
+                {
+                    $TokenSet = @{
+                        U = [Char[]]'ABCDEFGHIJKLMNOPQRSTUVWXYZ'
+                        L = [Char[]]'abcdefghijklmnopqrstuvwxyz'
+                        N = [Char[]]'0123456789'
+                        S = [Char[]]'!"#$%&''()*+,-./:;<=>?@[\]^_`{|}~'
+                    }
 
-                $Upper = Get-Random -Count 5 -InputObject $TokenSet.U
-                $Lower = Get-Random -Count 5 -InputObject $TokenSet.L
-                $Number = Get-Random -Count 5 -InputObject $TokenSet.N
-                $Special = Get-Random -Count 5 -InputObject $TokenSet.S
+                    $Upper = Get-Random -Count 5 -InputObject $TokenSet.U
+                    $Lower = Get-Random -Count 5 -InputObject $TokenSet.L
+                    $Number = Get-Random -Count 5 -InputObject $TokenSet.N
+                    $Special = Get-Random -Count 5 -InputObject $TokenSet.S
 
-                $StringSet = $Upper + $Lower + $Number + $Special
+                    $StringSet = $Upper + $Lower + $Number + $Special
 
-                $stringPassword = (Get-Random -Count 15 -InputObject $StringSet) -join ''
-                $passwordValue = ConvertTo-SecureString $stringPassword -AsPlainText -Force
+                    $stringPassword = (Get-Random -Count 15 -InputObject $StringSet) -join ''
```

Comment:
We create a 30 character password above. We should do the same here.

Reply:
i intentionally didn't touch this code as it was already in the module and didn't want to commit changes out of scope of the original issue (#3093)

in my opinion the resource should only use one method to generate passwords and [system.web.security.membership] should be removed. i can prepare the code change and open a separate issue/pr for this.